drivers/gpio/gpio_si32_hd.c

```diff
@@ -0,0 +1,194 @@
+/*
```

Comment:
I would suggest using a DT property to differentiate between high and non-high drive support, potentially incorporating Zephyr GPIO flags in a better way to enable that functionality, and combining these two gpio drivers.

There should be several drivers that can be used for reference. 

Additionally, I just wanted to make sure that there is no potential overlap with the gecko silabs gpio drivers.

@fkokosinski might know as silabs maintainer.

Reply:
> I would suggest using a DT property to differentiate between high and non-high drive support, potentially incorporating Zephyr GPIO flags in a better way to enable that functionality, and combining these two gpio drivers.

During development I though about that but decided against it because the registers look completely different and have different data types it would end up in every function needing an if/else block. I didn't like that because it wouldn't save any lines of code, would make the code harder to follow and (less importantly) reduce performance since you have to do that check for every gpio API call.